### ISSUE TITLE
[PHP] unable to access returned object due to protected container

### ISSUE THREAD

**Issue by jbs1** [swagger-api/swagger-codegen#6777]
I am using an php client generated with v.2.2.2 from this [json](https://esi.tech.ccp.is/latest/swagger.json).
When I call an some endpoint it would return something like eg. this:

```php
Swagger\Client\Model\GetCharactersCharacterIdContracts200Ok Object
(
    [container:protected] => Array
        (
            [acceptor_id] => 0
            [assignee_id] => 2112925746
            [availability] => personal
        (
)
``` 
But as you can see the container is 'protected', thus i can't access anything inside, which makes it pretty useless.

I don't know if the 'protected' is there on purpose or if it is an error with the generation?
If is is there on purpose, how else can i access the things inside?

**Comment by wing328** [comment]
Please try the latest master (2.3.0), in which the PHP client has been refactored with lots of improvements. SNAPSHOT version available in the README.

**Comment by jbs1** [comment]
when i try to use v2.3.0 according to the documentation generated it gives me an `Uncaught Error: Class Http\\Adapter\\Guzzle6\\Client`

**Comment by wing328** [comment]
cc @jebentier @dkarlovi @mandrean @jfastnacht @ackintosh

**Comment by ackintosh** [comment]
@jbs1  Hi. Thanks for reporting the issue.

- The first argument of API  is [optional](https://github.com/swagger-api/swagger-codegen/blob/master/samples/client/petstore/php/SwaggerClient-php/lib/Api/PetApi.php#L66).
- If the argument is omitted,  [`GuzzleHttp\Client` will be instantiate](https://github.com/swagger-api/swagger-codegen/blob/master/samples/client/petstore/php/SwaggerClient-php/lib/Api/PetApi.php#L70) in its constructor.

Therefore, the parameter `new \Http\Adapter\Guzzle6\Client()` which in generated document, is wrong.
We should fix it. 💡

**Comment by jbs1** [comment]
without using the `new \Http\Adapter\Guzzle6\Client()`
it leads to failing of methods that need OAuth authentication by throwing a 
`resulted in a '403 Forbidden' response:\n{"error": "Missing or invalid token.", "sso_status": 400}`

and it doesn't matter whether I directly give the access token as a parameter in the method or set it via
`Swagger\Client\Configuration::getDefaultConfiguration()->setAccessToken('YOUR_ACCESS_TOKEN');`

**Comment by ackintosh** [comment]
@jbs1 The matter about `Guzzle6\Client` has been fixed by #6930. ✨

**Comment by wing328** [comment]
@jbs1 I think you should access the properties using the getter instead. Here is an example extracted from the petstore integration tests:
```
    /**
     * Test case for getPetById
     *
     * Find pet by ID.
     *
     */
    public function testGetPetById()
    {
        // initialize the API client without host
        $pet_id = 10005;  // ID of pet that needs to be fetched

        $config = new Configuration();
        $config->setApiKey('api_key', '111222333444555');
        $pet_api = new Api\PetApi(null, $config);
        // return Pet (model)
        $response = $pet_api->getPetById($pet_id);
        $this->assertSame($response->getId(), $pet_id);
        $this->assertSame($response->getName(), 'PHP Unit Test');
        $this->assertSame($response->getPhotoUrls()[0], 'http://test_php_unit_test.com');
        $this->assertSame($response->getCategory()->getId(), $pet_id);
        $this->assertSame($response->getCategory()->getName(), 'test php category');
        $this->assertSame($response->getTags()[0]->getId(), $pet_id);
        $this->assertSame($response->getTags()[0]->getName(), 'test php tag');
    }
```
